### ISSUE TITLE
Ability to provide a SPARQL query template on each property or class in the config to setup the widget datasource

### ISSUE THREAD

**Issue by tfrancart** [sparna-git/Sparnatural#153]

**Comment by lubianat** [comment]
Is this possible?

**Comment by tfrancart** [comment]
Yes:

1. See ds:datasource in OWL-based configuration : https://github.com/sparna-git/Sparnatural/wiki/OWL-based-configuration#properties-configuration-reference - but this is not fully documented yet (more documentation will come soon)
1. Here for a JSON-based configuration : https://github.com/sparna-git/Sparnatural/wiki/JSON-based-configuration#reference-to-a-preconfigured-sparql-query--a-property or https://github.com/sparna-git/Sparnatural/wiki/JSON-based-configuration#your-own-sparql-query (but I don't recommend JSON-based configuration)

Basically you can design any sparql query to populate your dropdown lists or autocomplete fields.

You can look at the datasource config ontology at http://data.sparna.fr/ontologies/sparnatural-config-datasources/index-en.html, and look at examples OWL config in one of the demos at https://github.com/sparna-git/sparnatural.eu/tree/main/demos.

Ask questions if you need help.

**Comment by lubianat** [comment]
Thank you @tfrancart ! I'll be working on that in the next few days. Congrats for the great tool and thank you for the reply, by the way!
